src/main/kotlin/no/nav/familie/ef/iverksett/økonomi/utbetalingsoppdrag/UtbetalingsoppdragGenerator.kt

```diff
@@ -25,6 +33,149 @@ object UtbetalingsoppdragGenerator {
      * @param[forrigeTilkjentYtelse] Forrige tilkjent ytelse, med fullstendig sett av andeler med id
      * @return Ny tilkjent ytelse med andeler med id'er, samt utbetalingsoppdrag
      */
+    fun lagTilkjentYtelseMedUtbetalingsoppdragNy(
+        nyTilkjentYtelseMedMetaData: TilkjentYtelseMedMetaData,
+        forrigeTilkjentYtelse: TilkjentYtelse? = null,
+        erGOmregning: Boolean = false,
+    ): TilkjentYtelse {
+        val stønadstype = nyTilkjentYtelseMedMetaData.stønadstype
+        val personIdent = nyTilkjentYtelseMedMetaData.personIdent
+        val ytelseType = stønadstype.tilYtelseType()
+        var counter = 0
+        fun nextId() = (++counter).toString()
+
+        val forrigeAndeler = (forrigeTilkjentYtelse?.andelerTilkjentYtelse ?: emptyList())
+            .map { it.tilAndelData(id = nextId(), personIdent = personIdent, ytelseType = ytelseType) }
+        val nyeAndelerPåId = nyTilkjentYtelseMedMetaData.tilkjentYtelse.andelerTilkjentYtelse.associateBy { nextId() }
+
+        val nyeAndeler = tilAndelData(nyeAndelerPåId, personIdent, ytelseType)
+        val forrigeSisteAndelIKjede = forrigeTilkjentYtelse?.sisteAndelIKjede
+        val sisteAndelPerKjede = mapTilSisteAndelPerKjede(forrigeSisteAndelIKjede, personIdent, ytelseType)
+
+        val beregnetUtbetalingsoppdrag = Utbetalingsgenerator().lagUtbetalingsoppdrag(
+            behandlingsinformasjon = behandlingsinformasjon(
+                nyTilkjentYtelseMedMetaData,
+                erGOmregning,
+                forrigeTilkjentYtelse,
+            ),
+            nyeAndeler = nyeAndeler,
+            forrigeAndeler = forrigeAndeler,
+            sisteAndelPerKjede = sisteAndelPerKjede,
+        )
+        val beregnedeAndelerPåId = beregnetUtbetalingsoppdrag.andeler.associateBy { it.id }
+        val gjeldendeAndeler = nyeAndelerPåId.entries.map {
+            if (it.value.harNullBeløp()) {
+                // TODO vi kan vurdere om vi bare skal filtrere vekk de etter att man har laget utbetalingsoppdraget, då vil de ikke bli sendt med som "forrige andeler" i neste utbetalingsoppdrag
+                it.value
+            } else {
+                val beregnetAndel = beregnedeAndelerPåId.getValue(it.key)
+                it.value.copy(
+                    periodeId = beregnetAndel.periodeId,
+                    forrigePeriodeId = beregnetAndel.forrigePeriodeId,
+                    kildeBehandlingId = UUID.fromString(beregnetAndel.kildeBehandlingId),
+                )
+            }
+        }
+        // TODO settes kanskje ikke kildeBehandlingId i ef-sak? Settes den då nå kanskje feil hvis man avkorter en periode?
```

Comment:
KildebehandlingId settes i ef-sak - blir den behandlingId-en som lagret ned en andel. 

Kan være litt upresis hvis man revurderer fra en dato, men ikke gjør endringer - da oppdaterer vi kildebehandlingid i ef-sak, men i utgangspunktet er den jo "lik" den tidligere andelen.

Reply:
Til konsistensavstemming så sendes den med, men den brukes egentlige ikke

For utbetalingsgeneratorn så blir den egentligen overskrevet for "nyopprettede andeler"